src/main/java/org/broadinstitute/hellbender/tools/exome/ExomeReadCounts.java

```diff
@@ -0,0 +1,681 @@
+
+    private int groupCount = 1;
+
+    private static final Logger logger = LogManager.getLogger(ExomeReadCounts.class);
+
+    private ExonDB<GenomeLoc> exonDB;
+
+    private Counts counts;
+
+    private GenomeLocParser genomeLocFactory;
+
+    @Override
+    public void onTraversalStart() {
+        super.onTraversalStart();
+        final SAMSequenceDictionary sequenceDictionary = getBestAvailableSequenceDictionary();
+        if (sequenceDictionary == null)
+            throw new UserException("missing reference sequence dictionary, as no input contains such a dictionary... please consider to provide fail-over dictionary using " + 
+                                    "a reference (e.g. --reference myref.fasta)");
+        if (!intervalArgumentCollection.intervalsSpecified())
+            throw new UserException("you must indicate the set of exon as input intervals");
+
+        genomeLocFactory = new GenomeLocParser(sequenceDictionary);
+        sampleDB = new SampleDB(getHeaderForReads());
+        logger.log(Level.INFO,"Reading exons locations from intervals...");
+        final List<GenomeLoc> genomeLocs =
+                fromSimpleIntervalsToGenomeLocs(genomeLocFactory,intervalArgumentCollection.getIntervals(sequenceDictionary));
+        exonDB = new IntervalBackedExonDB(genomeLocs);
+        logger.log(Level.INFO,String.format("   Found %d exons.", exonDB.exonCount()));
+        logger.log(Level.INFO,"...done.");
+        counts = createCounts();
+        outputWriter = openOutputWriter(output);
+        if (summaryPerColumnOutput != null)
+            summaryPerColumnOutputWriter = openOutputWriter(summaryPerColumnOutput);
+        if (summaryPerRowOutput != null)
+            summaryPerRowOutputWriter = openOutputWriter(summaryPerRowOutput);
+
+        logger.log(Level.INFO,"Calculating exon depths...");
+    }
+
+    private List<GenomeLoc> fromSimpleIntervalsToGenomeLocs(final GenomeLocParser genomeLocFactory,
+                                                            final List<SimpleInterval> intervals) {
+        if (intervals == null || intervals.isEmpty())
+            return Collections.emptyList();
+        else if (intervals.size() == 1) {
+            final Locatable locatable = intervals.get(0);
+            return Collections.singletonList(genomeLocFactory.createGenomeLoc(locatable.getContig(),locatable.getStart(),locatable.getEnd()));
+        } else {
+            final List<GenomeLoc> result = new ArrayList<>(intervals.size());
+            for (final Locatable locatable : intervals) {
+                if (locatable == null)
+                   throw new IllegalArgumentException("null interval provided by the engine??? " + Arrays.toString(intervals.toArray()));
+                result.add(genomeLocFactory.createGenomeLoc(locatable.getContig(),locatable.getStart(),locatable.getEnd()));
+            }
+            return result;
+        }
+    }
+
+    private Counts createCounts() {
+        switch(groupBy) {
```

Comment:
the fact that there this enum exists and it has the same items as the 'Counts' hierarchy is a bad code smell. It's duplicating the logic of saying that there are 3 ways of grouping coverage counts.